server/command/command.go

```diff
@@ -49,16 +49,75 @@ func NewHandler(args *model.CommandArgs, conf *config.Config, a splunk.Splunk) H
 
 // GetSlashCommand returns command to register
 func GetSlashCommand() *model.Command {
+	splunk := model.NewAutocompleteData(
+		slashCommandName, "[alert|auth|help|log]", "connect to and interact with splunk.")
+	addSubCommands(splunk)
+
 	return &model.Command{
 		Trigger:          slashCommandName,
 		DisplayName:      slashCommandName,
 		Description:      pluginDescription,
 		AutoComplete:     true,
+		AutocompleteData: splunk,
 		AutoCompleteDesc: autoCompleteDescription,
 		AutoCompleteHint: autoCompleteHint,
 	}
 }
 
+func addSubCommands(splunk *model.AutocompleteData) {
+	splunk.AddCommand(createAlertCommand())
+	splunk.AddCommand(createAuthCommand())
+	splunk.AddCommand(createHelpCommand())
+	splunk.AddCommand(createlogCommand())
+}
+
+func createAlertCommand() *model.AutocompleteData {
+	alert := model.NewAutocompleteData(
+		"alert", "--subscribe", "subscribe to alert")
+
+	flag := []model.AutocompleteListItem{
+		{HelpText: "Subscribe", Item: "--subscribe"},
+	}
+	alert.AddStaticListArgument("Subscribe to alert", true, flag)
+	return alert
+}
+
+func createAuthCommand() *model.AutocompleteData {
+	auth := model.NewAutocompleteData(
+		"auth", "--login [server base url] [username/token]", "log into the splunk server")
+
+	flag := []model.AutocompleteListItem{
+		{HelpText: "Login to splunk server", Item: "--login"},
+	}
+
+	auth.AddStaticListArgument("Login to splunk server [server base url] [username/token]", true, flag)
+	auth.AddTextArgument("[server base url]", "Enter the server URL, e.g. https://your-mattermost-url.com", "")
+	auth.AddTextArgument("[username/token]", "Enter the [username/token]", "")
+
+	return auth
+}
+
+func createHelpCommand() *model.AutocompleteData {
+	help := model.NewAutocompleteData(
+		"help", "Display slash command help text", "")
+
+	return help
+}
+
+func createlogCommand() *model.AutocompleteData {
+	log := model.NewAutocompleteData(
+		"log", "[--list / logname]", "")
+
+	flag := []model.AutocompleteListItem{
+		{HelpText: "list all the log group", Item: "--list"},
+	}
+
+	log.AddStaticListArgument("list all the log group", false, flag)
```

Comment:
Other `staticListArguments` are all capitalised. Can we make it consistent. Made all of them capitalised or "lowercased".  I don't have preference.

Reply:
I agree, anything that reads like a sentence (a description of a field or a hint) should be capitalized

Reply:
@bakurits Resolved the changes